src/client/entity-editor/common/validation-label.tsx

```diff
@@ -16,13 +16,14 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
-// @flow
 
 import * as React from 'react';
 import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
 
 
-function icon(empty: ?boolean, error: ?boolean, warn: ?boolean): string | null {
+type OptionalBool = boolean | null | undefined;
+type Icon = 'times' | 'exclamation-triangle' | 'check';
+function icon(empty: OptionalBool, error: OptionalBool, warn: OptionalBool): Icon | null {
```

Comment:
```suggestion
function icon(empty?: boolean, error?: boolean, warn?: boolean): Icon | null {
```

Here again I wonder if creating an OptionalBool type is necessary when `xyz?: boolean` is an optional bool.

Reply:
Did a grep for icon, and it looks like it's only used inside `ValidationLabel`, where all three parameters are always provided in each call. I think the `?boolean` used to be there to indicate that the values could be null or undefined, because they come from the ValidationLabel props, where they are optional. Optional props will get destructured to `undefined`. I'd be happy to change this to `boolean | undefined` if you think that's better than what we have.

Reply:
No need, after all it works fine, is readable and declarative.